### ISSUE TITLE
Start global transactions for some special SQLs

### ISSUE THREAD

**Issue by djshow832** [pingcap/tidb#20853]
## Development Task

Some special SQLs visit metadata or they are executed in background sessions, so they should always begin as a global transaction.
This is a part of the global/local transaction roadmap #20448.

## Rationale

The special SQLs mentioned above include:
- Statements that write metadata, e.g. DDL, select nextval(seq), insert with an auto-increment column. Obtaining the next value from an auto-increment column or sequence may trigger an extra transaction, which writes the metadata to update the next start value. These statements create new KV transactions to write metadata, which is global and cannot be bound to any DC, so we need to guarantee that the new transactions are global.
- Statements(except for DMLs) that write system tables, e.g. analyze tables, grant privileges, set global variables, create bindings. These statements create new transactions in background system sessions to write system tables. a.k.a. internal SQLs. We need to guarantee that the new transactions are global.
- DDLs that write user tables, e.g. create/drop indexes, add columns, recover/cleanup indexes. Creating indexes and adding columns do the backfill jobs in background worker goroutines. Recover indexes and cleanup indexes do the backfill jobs in a user session. They all create new KV transactions to write user tables. We need to guarantee that all the new transactions are global.
- DMLs that write system tables, e.g. update mysql.tidb directly. These statements won't create new transactions, but they will fail as described in #20827. It's acceptable.

To summarize, we need to consider 3 scenarios:
- Internal SQLs. They always run in system sessions.
- SQLs that write metadata. They may run in user sessions.
- DDLs that write user tables. They may run in user sessions.

## Implementations

Internal SQLs can be recognized by `SessionVars.InRestrictedSQL` or `session.isInternal()`. Internal statements are executed through `session.ExecuteInternal()` or `execRestrictedSQL()`.

Transactions that write metadata are executed through `Meta.txn` API. We need to guarantee that the `kv.Transaction` passed to `NewMeta` is started as a global transaction.

DDLs that write user tables can be found in `RecoverIndexExec`, `CleanupIndexExec`, `addIndexWorker`, `cleanUpIndexWorker`, `updateColumnWorker`, etc.
